en/codegen/layout-codegen.md

```diff
@@ -0,0 +1,375 @@
+| <img src="../images/layout_codegen_table1.png" /> | Known as the “default” state. When an element is a single child in a row, width will be 100%. When the element is in a row with multiple elements, the below proportional width sizing rules will be applied. |
+| <img src="../images/layout_codegen_table2.png" /> | Width will be fixed.                                                                                                                                                                                          |
+| <img src="../images/layout_codegen_table3.png" /> | When an element is a single child in a row, its width will be 100%. When the element is in a row with multiple elements, the below proportional width sizing rules will be applied.                           |
+
+### Sizing Exceptions
+
+The following controls will render using the default width provided by the control’s HTML unless the width is fixed or pinned left and right:
+
+- Badge
+- Bottom Nav
+- Button
+- Button Group
+- Calendar
+- Card
+- Carousel
+- Checkbox
+- Dialog
+- Hyperlink
+- Linear Progress
+- Nav Bar
+- Nav Drawer
+- Paragraph
+- Radio
+- Slider
+- Switch
+- Time Picker
+- Title
+- Typography
+
+The following controls will always render using the default width provided by the control’s HTML, irrespective of any Sketch sizing or pinning:
+
+- Avatar
+- Circular Progress
+- Icon
+- Toast
+
+Snackbar is always rendered with a width of 100%.
+Category Chart width is always fixed to match with width in Sketch. This will be changed in the future to accomodate improvements in Ignite UI for Angular.
+Image always uses fixed width from Sketch.
+The Grid is always rendered with a fixed width equal to the sum of the column widths.
+
+### Fixed Sizing
+
+For elements except for the above exceptions, when width is fixed, the width will be fixed.
+
+### Pinned Left and Right Sizing
+
+For elements except for the above exceptions, when pinned left and right width will be 100%.
+
+### Proportional Width Sizing
+
+For elements except for the above exceptions, proportional width sizing applies to the element when the width is not fixed.
+If the element is the only element in the row, it will be sized width 100%.
+If the row the element is in has a fixed width, the element will be sized using a fixed width.
+For this Beta, all proportional width values will be 100% and the developer can make necessary adjustments.
+
+## Group Width
+
+When a Sketch group has a fixed width, that width value will be applied, otherwise no width value will be assigned in CSS.
+
+## Alignment
+
+### Centering
+
+The below centering logic only applies to a Sketch Group and no other elements. Further, the Group must be a child of the Artboard or Group that was selected for code generation.  
+All groups that are set up to center must have content. Content could be another group or any other element, except a background color, a background color does not count as an element.
```

Comment:
The last part should be its own sentence, as currently it is a run-on (incorrect) sentence.  "A background color does not count as an element." or "except a background color, because a background color...".